examples/cdecl.py

```diff
@@ -95,6 +116,65 @@ def _explain_type(decl):
         return ('function(%s) returning ' % (args) +
                 _explain_type(decl.type))
 
+    elif typ == c_ast.Struct:
+        decls = [_explain_decl_node(mem_decl) for mem_decl in decl.decls]
+        members = ', '.join(decls)
+
+        return ('struct%s ' % (' ' + decl.name if decl.name else '') +
+                ('containing {%s}' % members if members else ''))
+
+
+def _expand_struct_typedef(decl, file_ast, expand_struct=False, expand_typedef=False):
+    """Recursively expand struct & typedef if according argument is true,
+       also check using undeclared struct & type and throw Exception
+    """
+    typ = type(decl)
+
+    if typ in (c_ast.Decl, c_ast.TypeDecl, c_ast.PtrDecl, c_ast.ArrayDecl):
```

Comment:
I have made it return a new decl. 

And I wonder that if expansion is only used in `explain_c_declaration`, then users won't have access to that node. Does it means that saying it "much more involved"  is based on the fact that it should be public to users and thus having the original node unchanged after expansion becomes a flexibility?

Just to check whether I get your idea and thanks in advance.